include/wx/animate.h

```diff
@@ -49,6 +57,13 @@ class WXDLLIMPEXP_CORE wxAnimationBase : public wxObject
     virtual bool Load(wxInputStream& stream,
                       wxAnimationType type = wxANIMATION_TYPE_ANY) = 0;
 
+    // extended interface used only by the generic implementation of wxAnimationCtrl
+    virtual wxPoint GetFramePosition(unsigned int frame) const = 0;
+    virtual wxSize GetFrameSize(unsigned int frame) const = 0;
+    virtual wxAnimationDisposal GetDisposalMethod(unsigned int frame) const = 0;
+    virtual wxColour GetTransparentColour(unsigned int frame) const = 0;
+    virtual wxColour GetBackgroundColour() const = 0;
+
```

Comment:
Sorry, I don't understand why did we have to move all these methods into the base class. It looks like they're only used in the generic code and generic code should always work with generic animation objects, so it really should be enough to have these methods there only.

Am I missing something?

Reply:
I was thinking that it might make sense to allow custom classes derived from `wxAnimationBase` in some (likely rare) cases. That doesn't make as much sense with the `Impl` model, and it would probably be better with custom decoder classes instead in either case.